src/reconstruction/community.jl

```diff
@@ -0,0 +1,409 @@
+their `exchange_rxn_ids` and `exchange_met_ids`. These exchange reactions and
+metabolites link to environmental metabolites and reactions.
+Optionally specify `species_names` to append a specific name to each reaction
+and metabolite of an organism for easier reference (default is `species_i` for
+each model index i in `models`). Note, the bounds of the environmental variables
+are all set to zero. Thus, to run a simulation you need to constrain them
+appropriately. All the other bounds are inherited from the models used to
+construct the community model.
+
+If `add_biomass_objective` is true then `biomass_ids` needs to be supplied as
+well. This creates a model with an extra reaction added to the end of the
+stoichiometric matrix (last column) that can be assigned as the objective
+reaction. It also creates biomass "metabolites" that can be used in this
+objective reaction. Note, this reaction is unspecified, further action needs to
+be taken to specify it, e.g. assign weights to the last column of the
+stoichiometric matrix in the rows corresponding to the biomass metabolites.
+
+To further clarify how this `join` works. Suppose you have 2 organisms with
+stoichiometric matrices S₁ and S₂ and you want to link them with
+`exchange_met_ids = [em₁, em₂, em₃, ...]` and `exchange_rxn_ids = [er₁, er₂,
+er₃, ...]`. Then a new community stoichiometric matrix is constructed that looks
+like this:
+```
+            _      er₁  er₂  er₃  ...  b_
+           |S₁                           |
+           |   S₂                        |
+        em₁|                             |
+S   =   em₂|                             |
+        em₃|                             |
+        ...|                             |
+        bm₁|                             |  
+        bm₂|_                           _|
+
+```
+The exchange reactions in each model get linked to environmental metabolites,
+`emᵢ`, and these get linked to environmental exchanges, `erᵢ`. These `erᵢ`
+behave like normal single organism exchange reactions. When
+`add_biomass_objective` is true each model's biomass becomes a pseudo-metabolite
+(`bmᵢ`). These can be weighted in column `b`, called the `community_biomass`
+reaction in the community model, if desired. Refer to the tutorial if this is
+unclear.
+
+# Example
+```jldoctest; output = false
+m1 = load_model(core_model_path)
+m2 = load_model(CoreModel, core_model_path)
+
+boundary_rxn_ids, boundary_met_ids = all_boundaries(m2)
+exchange_rxn_ids = filter(startswith("EX_"), boundary_rxn_ids)
+exchange_met_ids = filter(endswith("_e"), boundary_met_ids)
+
+biomass_ids = ["BIOMASS_Ecoli_core_w_GAM", "BIOMASS_Ecoli_core_w_GAM"]
+
+community = COBREXA.join(
+    [m1, m2],
+    exchange_rxn_ids,
+    exchange_met_ids;
+    add_biomass_objective = true,
+    biomass_ids = biomass_ids,
+)
+
+size(stoichiometry(community))
+
+# output
+
+(166, 211)
+```
+"""
+function Base.join(
```

Comment:
the name `join` may be a bit too generic here, moreover it's in a base. It doesn't say that we're joining the models in a way that makes a community with exchange reactions, and there might be other ways.

Better ideas:
- `make_community_with_exchanges`
- `join_with_exchanges`